### PR TITLE
Make Doc API Doc

### PR DESCRIPTION
**Pull request by mhoeber**

### CODE REVIEW

**Comment by mhoeber** [comment]
@nedbat   @cpennington  this version is in progress:  http://xblock.readthedocs.org/en/markhoeber-make-doc-api-only/

I need to check and test it tomorrow.

File: doc/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.. _Introduction to XBlocks:
+
+#############################
+Introduction to XBlocks
+#############################
+
+You use the XBlocks APIs to build components that course teams can use to
+create independent course components that work seamlessly with other components
+in an online course.
```
Comment (nedbat) [inline doc/introduction.rst]: "You build components that course teams use to create components", this seems confusing.

File: doc/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+For example, you can build XBlocks to represent individual problems or pieces
+of text or HTML content. Furthermore, like Legos, XBlocks are composable; you
+can build XBlocks to represent larger structures such as lessons, sections, and
+entire courses.
+
+A primary advantage to XBlocks is that they are deployable. The code you write
```
Comment (nedbat) [inline doc/introduction.rst]: I don't understand what deployable means here, in contrast to any other code inside edx-platform.  "Sharable"?

File: doc/introduction.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+can be deployed in any instance of the edX Platform or other XBlock runtime
+application, then used by any course team using that system.
+
+By combining XBlocks from a wide variety of sources, from text and video, to
+multiple choice and numerical questions, to sophisticated collaborative and
+interactive learning laboratories, course teas can create rich and engaging
```
Comment (nedbat) [inline doc/introduction.rst]: "teas" --> "teams"

File: doc/introduction.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+courseware.
+
+*****************************************
+XBlock Independence and Interoperability
+*****************************************
+
+You must design your XBlock to meet two goals.
+
+* The XBlock must be independent of other XBlocks. Course teams must be able to
+  use the XBlock without using other XBlocks.
```
Comment (nedbat) [inline doc/introduction.rst]: "without using other XBlocks": every course will use other XBlocks.

**Comment by mhoeber** [comment]
@cpennington   addressed ned's comments.  OK to merge/publish simultaneously with tutorial?

**Comment by cpennington** [comment]
:+1:

**Pull request merged**